ssl/ssl_lib.c

```diff
@@ -4516,6 +4516,37 @@ size_t SSL_early_get0_compression_methods(SSL *s, const unsigned char **out)
     return s->clienthello->compressions_len;
 }
 
+int SSL_early_get1_extensions_present(SSL *s, int **out, size_t *outlen)
+{
+    RAW_EXTENSION *ext;
+    size_t num = 0, i;
+
+    if (s->clienthello == NULL || out == NULL || outlen == NULL)
+        return 0;
+    for (i = 0; i < s->clienthello->pre_proc_exts_len; i++) {
+        ext = s->clienthello->pre_proc_exts + i;
+        if (ext->present)
+            num++;
+    }
+    *out = OPENSSL_malloc(sizeof(**out) * num);
```

Comment:
I actually prefer that style (only writing to output variables on success) but didn't think that we had consistent usage of it in openssl yet.  I'm happy to make that change, but probably not until after I arrive at the IETF meeting.